src/Symfony/Component/Console/Output/StreamOutput.php

```diff
@@ -92,12 +92,16 @@ protected function doWrite($message, $newline)
      */
     protected function hasColorSupport()
     {
-        // @codeCoverageIgnoreStart
-        if (DIRECTORY_SEPARATOR === '\\') {
-            return false !== getenv('ANSICON') || 'ON' === getenv('ConEmuANSI') || 'xterm' === getenv('TERM');
+        if (DIRECTORY_SEPARATOR == '\\') {
+            return false !== getenv('ANSICON') || 'ON' === getenv('ConEmuANSI');
+        } elseif (!function_exists('posix_isatty')){
+            return false;
         }
 
-        return function_exists('posix_isatty') && @posix_isatty($this->stream);
-        // @codeCoverageIgnoreEnd
+        try {
+            return posix_isatty($this->stream);
```

Comment:
Do you recommend to close this pull request and to declare the related issue as "won't fix"? Thanks.